src/networking.c

```diff
@@ -2916,9 +2917,26 @@ int tio_debug = 0;
 #define IO_THREADS_OP_READ 0
 #define IO_THREADS_OP_WRITE 1
 
+typedef struct IOPending_t {
+    redisAtomic unsigned long var;
+    /* Mutexes used to protect atomic variables when atomic builtins are
+     * not available. */
+    pthread_mutex_t var_mutex;
+} IOPending_t;
```

Comment:
Not sure why we don't use this abstraction everywhere? We could have a typedef struct for each type: redisAtomic_size_t, redisAtomic_long_long, etc and use the same inline functions to access them. It would prevent using them in a non-thread safe way. 

Is there some performance issue or other tradeoff?

Reply:
posix mutexs are ridiculously slow.  Seems not a good idea to protect atomic vars with them as a fallback since perf is just destroyed.

Reply:
@madolson I want to follow salvatore's design(described on his comments) and don't want to make it less readable. There must be many micros if we support redisAtomic_size_t, redisAtomic_long_long, etc,. even we can introduce a third-party library, but i think that will make it complicated.

@JohnSully Yes, I clearly know `pthread_mutex` has low performance, but, just like before, i have no idea what we can use if compilers doesn't support atomic operations.  And i think it is a relatively rare case, maybe using compiler with a low version.

Reply:
I would highly recommend falling back to the GCC atomic builtins which exist on most platforms. They also work in clang and a few other compilers.

Reply:
hi @JohnSully I think you may miss something. As src/src/atomicvar.h said,  if c11 atomic is not supported, we use gcc atomic, and if gcc atomic is not supported, we use sync operation. Finally, only if sync operation is not supported, we use `phtread_mutex`.